test/functional/legacy/079_substitute_special_spec.lua

```diff
@@ -0,0 +1,377 @@
+-- Test for *sub-replace-special* and *sub-replace-expression* on :substitute.
+-- Test for submatch() on :substitue.
+-- Test for *:s%* on :substitute.
+-- Test for :s replacing \n with  line break.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+local eq, eval, write_file = helpers.eq, helpers.eval, helpers.write_file
+
+local function expect_line(line)
+  return eq(line, eval('getline(".")'))
+end
+
+describe(':substitue', function()
+  setup(function()
+    write_file('test_79_7.in', 'TEST_7:\nA\rA\nB\x00B\nC\x00C\nD\x00\nD\nE'..
+      '\x00\n\x00\n\x00\n\x00\n\x00E\nQ\nQ\n')
+  end)
+  before_each(clear)
+  teardown(function()
+    os.remove('test_79_7.in')
+  end)
+
+  it('with "set magic"', function()
+    insert([[
+      TEST_1:
```

Comment:
perhaps the same thing that was done in test80 could be done here, where the `TEST_N` text could be moved into the first argument of `it()`.